src/test/java/coursera/services/StudentServiceTest.java

```diff
@@ -0,0 +1,69 @@
+package coursera.services;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+import coursera.entities.Student;
+import coursera.reports.Reporter;
+import coursera.repositories.StudentRepository;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class StudentServiceTest {
+    StudentRepository studentRepository;
+    Reporter reporter;
+    StudentService studentService;
+
+    @BeforeAll
+    public void setup() {
+        reporter = mock(Reporter.class);
+        studentRepository = mock(StudentRepository.class);
+        studentService = new StudentService(studentRepository, reporter);
+    }
+
+    @Test
+    void GetReport_ByCredit_WithoutStudents() {
+        // GIVEN
+        when(studentRepository
+                .findAllStudentsByCredit(50, "2020-10-08", "2022-10-08"))
+                .thenReturn(null);
+        // WHEN
+        studentService.getReport("html", new String[]{}, 50, "2020-10-08", "2022-10-08");
+        // THEN
+        verify(reporter).writeReportToFile("html", null);
+    }
+
+    @Test
+    void GetReport_ByCredit_ReturnException() {
+        when(studentRepository
+                .findAllStudentsByCredit(20, "2020-01-12", "2019-08-30"))
+                .thenThrow(new RuntimeException());
```

Comment:
It's easier to reach if there is a single empty line between the different sections of the unit test (given, when, then)